src/lib/core/TLVCircularBuffer.cpp

```diff
@@ -194,14 +194,34 @@ CHIP_ERROR TLVCircularBuffer::OnInit(TLVWriter & writer, uint8_t *& bufStart, ui
 
 CHIP_ERROR TLVCircularBuffer::GetNewBuffer(TLVWriter & ioWriter, uint8_t *& outBufStart, uint32_t & outBufLen)
 {
-    uint8_t * tail = QueueTail();
-
     if (mQueueLength >= mQueueSize)
     {
         // Queue is out of space, need to evict an element
         ReturnErrorOnFailure(EvictHead());
     }
 
+    return GetCurrentBuffer(ioWriter, outBufStart, outBufLen);
+}
+
+/**
+ * @brief
+ *   returns the actual state of what our current available buffer space is 
+ *
+ * @param[in,out] ioWriter  TLVWriter calling this function
+ *
+ * @param[out] outBufStart The pointer to the new buffer
+ *
+ * @param[out] outBufLen   The available length for writing
+ *
+ * @retval #CHIP_NO_ERROR On success.
+ *
+ * @retval other           If the function was unable to elide a complete
+ *                         top-level TLV element.
```

Comment:
This doesn't make any sense.  This function never "elide"s anything, and always succeeds....

Reply:
updated, thanks